src/theory/builtin/generic_op.cpp

```diff
@@ -0,0 +1,320 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2022 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * A class for representing abstract types.
+ */
+
+#include "theory/builtin/generic_op.h"
+
+#include <iostream>
+
+#include "expr/dtype.h"
+#include "expr/dtype_cons.h"
+#include "theory/datatypes/theory_datatypes_utils.h"
+#include "util/bitvector.h"
+#include "util/floatingpoint.h"
+#include "util/iand.h"
+#include "util/rational.h"
+#include "util/regexp.h"
+
+using namespace cvc5::internal::kind;
+
+namespace cvc5::internal {
+
+std::ostream& operator<<(std::ostream& out, const GenericOp& op)
+{
+  return out << "(GenericOp " << op.getKind() << ')';
+}
+
+size_t GenericOpHashFunction::operator()(const GenericOp& op) const
+{
+  return kind::KindHashFunction()(op.getKind());
+}
+
+GenericOp::GenericOp(Kind k) : d_kind(k) {}
```

Comment:
Is the plan to use this class only for indexed kinds or also for regular kinds?

Reply:
We only construct these for indexed kinds currently, but there isn't any harm in constructing one for a regular kind.